### PR TITLE
Access control in cloud functions

### PR DESCRIPTION
**Pull request by thompchr**

### CODE REVIEW

**Review by clarah2 — APPROVED** [review]

**Comment by clarah2** [comment]
@thompchr thanks for this! I'll let you resolve the conflicts and merge :)

**Comment by thompchr** [comment]
@clarah2  Can you take a quick look at this and re-review, now that I've merged it? Once that's done you should be able to merge into dev

**Review by clarah2 — APPROVED** [review]

**Pull request merged**
